src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs

```diff
@@ -0,0 +1,160 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	[Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerDnsAlias", SupportsShouldProcess = true)]
+	[OutputType(typeof(Model.AzureSqlServerDnsAliasModel))]
+	public class RemoveAzureSqlServerDNSAlias : AzureSqlServerDnsAliasCmdletBase
+	{
+		protected const string CmdletInputParameterSet =
+			"Remove a Server Dns Alias from cmdlet input parameters";
+
+		protected const string ObjectParameterSet =
+			"Remove a Server Dns Alias from AzureSqlServerDnsAliasModel instance definition";
+
+		protected const string ResourceIdParameterSet =
+			"Remove a Server Dns Alias from an Azure resource id";
+
+		/// <summary>
+		/// Gets or sets the name of the server dns alias to remove
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletInputParameterSet,
+			Mandatory = true,
+			HelpMessage = "Azure Sql Server Dns Alias name")]
+		[Alias("DnsAliasName")]
+		[ValidateNotNullOrEmpty]
+		public string Name { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server to use
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletInputParameterSet,
+			Mandatory = true,
+			HelpMessage = "The Azure Sql Server name.")]
+		[ValidateNotNullOrEmpty]
+		public override string ServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the resource group to use.
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletInputParameterSet,
+			Mandatory = true,
+			Position = 0,
+			ValueFromPipelineByPropertyName = true,
+			HelpMessage = "The name of the resource group.")]
+		[ValidateNotNullOrEmpty]
+		public override string ResourceGroupName { get; set; }
+
+		/// <summary>
+		/// Server Dns Alias object to remove
+		/// </summary>
+		[Parameter(ParameterSetName = ObjectParameterSet,
+			Mandatory = true,
+			ValueFromPipeline = true,
+			HelpMessage = "The Server Dns Alias object to remove")]
+		[ValidateNotNullOrEmpty]
+		public Model.AzureSqlServerDnsAliasModel InputObject { get; set; }
+
+		/// <summary>
+		/// Gets or sets the resource id of the server dns alias
+		/// </summary>
+		[Parameter(ParameterSetName = ResourceIdParameterSet,
+			Mandatory = true,
+			HelpMessage = "The resource id of Server Dns Alias object to remove")]
+		[ValidateNotNullOrEmpty]
+		public string ResourceId { get; set; }
```

Comment:
@maboja-msft `ResourceId` needs to have the `ValueFromPipelineByPropertyName = true` property